blocks/autocomplete/index.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * External dependencies
+ */
+import { clone } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { applyFilters, hasFilter } from '@wordpress/hooks';
+import { Component } from '@wordpress/element';
+import { Autocomplete as OriginalAutocomplete } from '@wordpress/components';
+
+/*
+ * Use one array instance for fallback rather than inline array literals
+ * because the latter may cause rerender due to failed prop equality checks.
+ */
+const completersFallback = [];
+
+/**
+ * Wrap the default Autocomplete component with one that
+ * supports a filter hook for customizing its list of autocompleters.
+ *
+ * Since there may be many Autocomplete instances at one time, this component
+ * applies the filter on demand, when the component is first focused after
+ * receiving a new list of completers.
+ *
+ * This function is exported for unit test.
+ *
+ * @param  {Function} Autocomplete Original component.
+ * @return {Function}              Wrapped component
+ */
+export function withFilteredAutocompleters( Autocomplete ) {
+	return class FilteredAutocomplete extends Component {
+		constructor() {
+			super();
+
+			this.state = { completers: completersFallback };
+
+			this.saveParentRef = this.saveParentRef.bind( this );
+			this.onFocus = this.onFocus.bind( this );
+		}
+
+		componentDidUpdate() {
+			const hasFocus = this.parentNode.contains( document.activeElement );
+
+			// This may trigger another render but only when the component has focus.
+			if ( hasFocus && this.hasStaleCompleters() ) {
```

Comment:
Can you explain why we need this additional check in here? It seems like having `onFocus` handler should be enough.

Reply:
It is technically possible for the `completers` prop to change while the component contains the focus. This check exists for that case, since `onFocus` will already have been called. I don't think that is likely to occur, but I'd rather just cover the case. Sound OK?

Reply:
I hope it’s good 😃It makes sense, might make sense to leave an inline comment.

Reply:
It's mostly about not assuming when props will be updated. Sure, I can add a comment. :)